docs/cookbook/adding-a-new-elasticsearch-index.md

```diff
@@ -0,0 +1,294 @@
+# Adding a New Elasticsearch Index
+
+In this cookbook, we will add a new elasticsearch index for categories, implement basic functions for data export, implement cron module, and support for partial export.
+
+## New elasticsearch mapping
+
+As a first step we need to define [elasticsearch mapping](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping.html) in `src/Resources/definition/category/` for all domains (e.g. for 2 domains with ID 1 and 2: `1.json`, `2.json`).
+
+## New CategoryIndex
+
+Create a class `CategoryIndex` in `src/Model/Category/Elasticsearch`. The class must extends class `AbstractIndex`.
+
+```php
+declare(strict_types=1);
+
+namespace App\Model\Category\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex;
+
+class CategoryIndex extends AbstractIndex
+{
+    public function getName(): string
+    {
+        // TODO: Implement getName() method.
+    }
+    public function getTotalCount(int $domainId): int
+    {
+        // TODO: Implement getTotalCount() method.
+    }
+    public function getExportDataForIds(int $domainId, array $restrictToIds): array
+    {
+        // TODO: Implement getExportDataForIds() method.
+    }
+
+    public function getExportDataForBatch(int $domainId,int $lastProcessedId,int $batchSize) : array
+    {
+        // TODO: Implement getExportDataForBatch() method.
+    }
+}
+```
+
+Register new index into `services.yml`
+
+```yaml
+    Shopsys\FrameworkBundle\Model\Category\Elasticsearch\CategoryIndex: ~
+```
+
+## Create new index into elasticsearch
+
+To create an index into elasticsearch we need to implement `getName()` method in `CategoryIndex`.
+The best practice is to define index name into constant due to later usage for obtaining data.
+
+```diff
++   public const INDEX_NAME = 'category';
++
+    public function getName(): string
+    {
+-       // TODO: Implement getName() method.
++       return static::INDEX_NAME;
+    }
+```
+
+So far it is the most minimalistic implementation.
+Now we are able to create an index in elasticsearch by running `./phing elasticsearch-index-create -D elasticsearc.index=category`.
```

Comment:
In this point, I get the following error:
```
There is no index "category" registered (uncaught exception) at
packages/framework/src/Component/Elasticsearch/Exception/ElasticsearchIndexException.php line 97 
while running console command `shopsys:elasticsearch:indexes-create`
```

and I had to manually tag the service with elasticsearch.index....
```
   App\Model\Category\Elasticsearch\CategoryIndex:
        tags: ['elasticsearch.index']
```

Reply:
This is weird. Did not help you cleaning cache? I guess it should help - also it might be mentioned here.

Reply:
Cleaning cache did not help.
I have `CategoryIndex extends AbstractIndex` and in `project-base/config/services.yaml` I registered this class as a service.
When I remove the tag from CategoryIndex from services.yaml, it fails on the error above.

Reply:
Fixed via https://github.com/shopsys/shopsys/pull/1622/commits/403af0cffe379157eecc3a018b482043eab43c46